src/pages/MuteerOverview/MuteerOverview.tsx

```diff
@@ -0,0 +1,398 @@
+import { Button, FieldInput, Heading } from '@pzh-ui/components'
+import {
+    MagnifyingGlass,
+    ArrowDownZA,
+    ArrowDownAZ,
+    EllipsisVertical,
+} from '@pzh-ui/icons'
+import { isBefore, isValid } from 'date-fns'
+import { FC, useEffect, useState } from 'react'
+import { Helmet } from 'react-helmet'
+import { useQueryClient } from 'react-query'
+import { useNavigate, Link } from 'react-router-dom'
+
+import {
+    AmbitiesRead,
+    BelangenRead,
+    BeleidskeuzesRead,
+    BeleidsregelsRead,
+    BeleidsprestatiesRead,
+    BeleidsmodulesRead,
+    BeleidsdoelenRead,
+    MaatregelenRead,
+    ThemasRead,
+    VerordeningenRead,
+} from '@/api/fetchers.schemas'
+import { ContainerMain } from '@/components/Container'
+import Dropdown from '@/components/Dropdown'
+import { LoaderCard } from '@/components/Loader'
+import PageSpecificNavBar from '@/components/PageSpecificNavBar'
+import { PopUpAddPolicyToModule } from '@/components/Popup'
+import { filteredDimensieConstants } from '@/constants/dimensies'
+import useAuth from '@/hooks/useAuth'
+import { getFetcherForType } from '@/utils/getFetchers'
+import { removePolicyFromModule } from '@/utils/removePolicyFromModule'
+
+/**
+ * A component to display all the objects from a specific dimension
+ * @param {Object} dimensieConstants - Contains the variables of the dimension
+ */
+
+type PossiblePolicyTypes =
+    | AmbitiesRead
+    | BelangenRead
+    | BeleidskeuzesRead
+    | BeleidsregelsRead
+    | BeleidsprestatiesRead
+    | BeleidsmodulesRead
+    | BeleidsdoelenRead
+    | MaatregelenRead
+    | ThemasRead
+    | VerordeningenRead
+
+interface MuteerOverviewProps {
+    dimensieConstants: filteredDimensieConstants
+    hideAddObject?: boolean
+}
+
+const MuteerOverview = ({
+    dimensieConstants,
+    hideAddObject,
+}: MuteerOverviewProps) => {
+    const { user } = useAuth()
+
+    const [filterQuery, setFilterQuery] = useState('')
+    const [isLoading, setIsLoading] = useState(true)
+    const [ascending, setAscending] = useState(true)
+    const [policyObjects, setPolicyObjects] = useState<PossiblePolicyTypes[]>(
+        []
+    )
+
+    const navigate = useNavigate()
+
+    const titleSingular = dimensieConstants.TITLE_SINGULAR
+    const titlePlural = dimensieConstants.TITLE_PLURAL
+    const createNewSlug = dimensieConstants.SLUG_CREATE_NEW
+    const overviewSlug = dimensieConstants.SLUG_OVERVIEW
+
+    const useGetLineage = getFetcherForType(titleSingular)
+    const { data: policyObjectsFromAPI } = useGetLineage()
+
+    /**
+     * When the component mounts, fetch the objects from the API and prepare state:
+     * 1. Sort based on ascending or descending
+     * 2. Filter out archived objects
+     * 3. Filter based on filter Query value
+     * 4. Set in state
+     */
+    useEffect(() => {
+        if (!policyObjectsFromAPI) return
+
+        const sortedAndFilteredPolicyObjects = policyObjectsFromAPI
+            .sort((a, b) =>
```

Comment:
@Aidenbuis Might be nice to create a separate utility for the sorting function

Reply:
Done! See commit 2b5e7ee00e